x-pack/apm-server/aggregation/spanmetrics/aggregator_test.go

```diff
@@ -76,28 +76,36 @@ func TestAggregatorRun(t *testing.T) {
 		serviceName string
 		agentName   string
 		destination string
+		targetType  string
+		targetName  string
 		outcome     string
 		count       float64
 	}
 
 	destinationX := "destination-X"
 	destinationZ := "destination-Z"
+	trgTypeX := "trg-type-X"
+	trgNameX := "trg-name-X"
+	trgTypeZ := "trg-type-Z"
+	trgNameZ := "trg-name-Z"
 	inputs := []input{
-		{serviceName: "service-A", agentName: "java", destination: destinationZ, outcome: "success", count: 2},
-		{serviceName: "service-A", agentName: "java", destination: destinationX, outcome: "success", count: 1},
-		{serviceName: "service-B", agentName: "python", destination: destinationZ, outcome: "success", count: 1},
-		{serviceName: "service-A", agentName: "java", destination: destinationZ, outcome: "success", count: 1},
-		{serviceName: "service-A", agentName: "java", destination: destinationZ, outcome: "success", count: 0},
-		{serviceName: "service-A", agentName: "java", outcome: "success", count: 1}, // no destination
-		{serviceName: "service-A", agentName: "java", destination: destinationZ, outcome: "failure", count: 1},
+		{serviceName: "service-A", agentName: "java", destination: destinationZ, targetType: trgTypeZ, targetName: trgNameZ, outcome: "success", count: 2},
+		{serviceName: "service-A", agentName: "java", destination: destinationX, targetType: trgTypeX, targetName: trgNameX, outcome: "success", count: 1},
+		{serviceName: "service-B", agentName: "python", destination: destinationZ, targetType: trgTypeZ, targetName: trgNameZ, outcome: "success", count: 1},
+		{serviceName: "service-A", agentName: "java", destination: destinationZ, targetType: trgTypeZ, targetName: trgNameZ, outcome: "success", count: 1},
+		{serviceName: "service-A", agentName: "java", destination: destinationZ, targetType: trgTypeZ, targetName: trgNameZ, outcome: "success", count: 0},
+		{serviceName: "service-A", agentName: "java", outcome: "success", count: 1},                                             // no destination or service target
+		{serviceName: "service-A", agentName: "java", targetType: trgTypeZ, targetName: trgNameZ, outcome: "success", count: 1}, // no destination
+		// {serviceName: "service-A", agentName: "java", destination: destinationZ, outcome: "success", count: 1},               // no service target, possible?
```

Comment:
I think this scenario may not be possible since the exit spans will always have a target service set. @SylvainJuge for confirmation

Reply:
In theory, every call to an external service must have the `resource` and `service.target` fields properly set, however it can be modified/removed through the agent API (at least in the case of the Java agent), so this scenario is possible, even if quite uncommon.

I think we can assume that if users need to explicitly modify/remove the destination and service target, it is either to customize it to their (niche) needs, or a work-around to override the default values that have been set by automatic instrumentation.

Reply:
@SylvainJuge In the case of no service target should the output have `target.type = ""` and `target.name=""` or should it be nil?

Reply:
As per the current apm-server codebase, for empty `resource` [the code doesn't process the span](https://github.com/elastic/apm-server/blob/4406146feaaa5c3d900d6edd62d43614b872fc6f/x-pack/apm-server/aggregation/spanmetrics/aggregator.go#L211-L213). For `target.{type, name}` I think we have 3 options:

1. Process them with empty fields
2. Process them with target set as nil
3. Don't process the span if `target.*` not set, similar to what is done with resource

I think 3 might not be a good approach as there is existing issue #7920 which talks about missing destination metrics due to `resource` fields not set in the input with OTEL sdk.

Reply:
I think in the case where no `service.target` fields are provided, they should be `nil` in the apm-server code, and not stored with empty values but absent from the stored document. If I'm not mistaken, this behavior should be consistent with what we currently do for the `span.destination.service.resource`  field, so I would advocate for option 3.

For #7920 , the issue comes from the fact that we don't infer an equivalent `resource` for OTLP intake, which makes the current implementation of metrics aggregation ignore them. I've added a small chapter on Otel compatibility in the specification https://github.com/elastic/apm/blob/6d1eaa24acf1147c2e98a4a327c43a296213ad0d/specs/agents/tracing-spans-service-target.md#otel-and-bridge-compatibility which should cover this case, so implementing this part might also help fix #7920 , let me know if there is anything unclear or not covered in this area.

Reply:
@SylvainJuge Thanks for the clarification. For the implementation of the spec for nil `service.target`, if we were to consider point `3` it would be incompatible with older agents which are sending only the `resource`, given this, should we go with approach `2` i.e. setting the target as `nil` if it is not provided?

Reply:
Agents will send spans with either:
1. `resource` for older agents
2. `resource` and `service.target.*` fields for newer agents
3. no `resource` nor `service.target.*` fields when span is not an exit span.

The behavior on apm-server should be respectively to:
1. store `resource` as-is, store `service.target.type` = `""` (empty string and `service.target.name` = `resource`
2. store `resource` and `service.target.*` fields as provided
3. `resource` = nil and `service.target.*` fields should not be present as well.